src/main/java/hudson/plugins/tfs/browsers/TeamSystemWebAccessBrowser.java

```diff
@@ -47,44 +47,49 @@ private String getBaseUrlString(ChangeSet changeSet) throws MalformedURLExceptio
         if (url != null) {
             baseUrl = DescriptorImpl.getBaseUrl(url);
         } else {
-            baseUrl = String.format("%s/", getServerConfiguration(changeSet)); 
+            String scmUrl = getServerConfiguration(changeSet);
+            if (scmUrl.endsWith("/")) {
+              baseUrl = scmUrl;
+            } else {
+              baseUrl = String.format("%s/", scmUrl);
+            }
         }
         return baseUrl;
     }
 
     /*
-     * http://tswaserver:8090/cs.aspx?cs=99
+     * Gets the link to a specific change set.
+     * E.g. http://tswaserver:8090/_versionControl/changeset/99
      */
     @Override
     public URL getChangeSetLink(ChangeSet changeSet) throws IOException {
-        return new URL(String.format("%scs.aspx?cs=%s", getBaseUrlString(changeSet), changeSet.getVersion()));
+      return new URL(String.format("%s_versionControl/changeset/%s",
+                                  getBaseUrlString(changeSet),
+                                  changeSet.getVersion()));
     }
 
     /*
-     * http://tswaserver:8090/view.aspx?path=$/Project/Folder/file.cs&cs=99
+     * Gets the link for a specific file in a change set.
      */
     public URL getFileLink(ChangeSet.Item item) throws IOException {
-        return new URL(String.format("%sview.aspx?path=%s&cs=%s", getBaseUrlString(item.getParent()), item.getPath(), item.getParent().getVersion()));
+      return new URL(String.format("%s_versionControl/changeset/%s#path=%s&_a=contents",
+                                   getBaseUrlString(item.getParent()),
+                                   item.getParent().getVersion(),
+                                   URLEncoder.encode(item.getPath(),"UTF-8")));

```

Comment:
Can we do better than string concat/format for URL path and query string components?  That way, we wouldn't have to worry about calling whether we need to call `URLEncoder#encode` or not; it would be taken care of by some `UrlBuilder` of some sort.  Or build the URL path with one class and the query string with another, using something like https://github.com/Microsoft/Git-Credential-Manager-for-Mac-and-Linux/blob/master/src/main/java/com/microsoft/alm/helpers/UriHelper.java#L30